rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -75,36 +76,18 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     // To be compatible with ROS 1, we construct a default filename of
     // the form ~/.ros/log/<exe>_<pid>_<milliseconds-since-epoch>.log
 
-    // First get the home directory.
-    const char * homedir = rcutils_get_home_dir();
-    if (homedir == nullptr) {
-      // We couldn't get the home directory; it is not really going to be
-      // possible to do logging properly, so get out of here without setting
-      // up logging.
-      RCUTILS_SET_ERROR_MSG("Failed to get users home directory");
-      return RCL_LOGGING_RET_ERROR;
-    }
-
-    // SPDLOG doesn't automatically create the log directories, so make them
-    // by hand here.
-    char name_buffer[4096] = {0};
-    int print_ret = rcutils_snprintf(name_buffer, sizeof(name_buffer), "%s/.ros", homedir);
-    if (print_ret < 0) {
-      RCUTILS_SET_ERROR_MSG("Failed to create home directory string");
-      return RCL_LOGGING_RET_ERROR;
-    }
-    if (!rcutils_mkdir(name_buffer)) {
-      RCUTILS_SET_ERROR_MSG("Failed to create user .ros directory");
-      return RCL_LOGGING_RET_ERROR;
+    char * logdir = nullptr;
+    rcl_logging_ret_t dir_ret = rcl_logging_get_logging_directory(allocator, &logdir);
+    if (RCL_LOGGING_RET_OK != dir_ret) {
+      // We couldn't get the log directory, so get out of here without setting up
+      // logging.
+      return dir_ret;
     }
 
-    print_ret = rcutils_snprintf(name_buffer, sizeof(name_buffer), "%s/.ros/log", homedir);
-    if (print_ret < 0) {
-      RCUTILS_SET_ERROR_MSG("Failed to create log directory string");
-      return RCL_LOGGING_RET_ERROR;
-    }
-    if (!rcutils_mkdir(name_buffer)) {
-      RCUTILS_SET_ERROR_MSG("Failed to create user log directory");
+    // SPDLOG doesn't automatically create the log directories, so create them
+    rcpputils::fs::path logdir_path(logdir);
+    if (!rcpputils::fs::create_directories(logdir_path)) {
+      RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING("Failed to create log directory: %s", logdir);
```

Comment:
We are leaking `logdir` on failure here.  The same goes several times below.

Reply:
fixed as part of dd0a9768e2fd4ad15c087c849c090e4e1364509b